### PR TITLE
Frontend: improve performance initial commit

### PR DESCRIPTION
**Pull request by maximeflamant**
Made small changes to enhance frontend performance. High CPU usage was primarily caused by re-rendering the entire device list every 100ms.

### CODE REVIEW

File: packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx [hunk 1]
```diff
@@ -60,11 +63,25 @@ export const BridgeDetailsPage = () => {
...
+              Auto-refresh in {(timer / 1000).toFixed(0)} seconds...
+            </Typography>
+          )}
+        </Box>
 
-      {devices && (
-        <DeviceList devices={devices} timer={timer} onRefresh={refreshNow} />
-      )}
+        {devices && (
+          <MemoizedDeviceList devices={devices} onRefresh={refreshNow} />
```
Comment (t0bst4r) [inline packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx:82]: As you moved the timer out of the device list, the `onRefresh` prop shouldn‘t be needed anymore.

File: packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx [hunk 2]
```diff
@@ -28,7 +31,7 @@ export const BridgeDetailsPage = () => {
     sleepSeconds: 9,
     startImmediate: true,
     callback: useCallback(() => setSeed(Date.now()), [setSeed]),
-    onTick: setTimer,
+    onTick: useCallback(setTimer, []),
```
Comment (t0bst4r) [inline packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx:34]: `setTimer` is a `useState`-setter. It should not be necessary to wrap it with `useCallback`.

File: packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
 import { useDevices } from "../../hooks/data/devices.ts";
 import { DeviceList } from "../../components/device/DeviceList.tsx";
 import { useTimer } from "../../hooks/timer.ts";
 import { IconButton, Stack, Typography } from "@mui/material";
 import Box from "@mui/material/Box";
-import { Edit } from "@mui/icons-material";
+import { Edit, Refresh } from "@mui/icons-material";
+
+const MemoizedBridgeDetails = memo(BridgeDetails);
+const MemoizedDeviceList = memo(DeviceList);
```
Comment (t0bst4r) [inline packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx:14]: `memo` prevents re-rendering if NO parameter changes. Object equality is checked by checking the object reference and not its content. When refreshed the device objects are whole new objects from the response. Even if the content is the same, it’s still a new object. Memo won’t have any effect.

File: packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx [hunk 4]
```diff
@@ -60,11 +63,25 @@ export const BridgeDetailsPage = () => {
         </IconButton>
       </Box>
 
-      <BridgeDetails bridge={bridge} />
+      <MemoizedBridgeDetails bridge={bridge} />
+
+      <Stack spacing={2}>
```
Comment (t0bst4r) [inline packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx:68]: Is this extra stack necessary? Can’t we use the parent stack somehow?

**Review by t0bst4r — CHANGES_REQUESTED** [review]

File: packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx (hunk 4 above)
Comment (maximeflamant) [reply packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx:68]: I could definitely use the parent stack and its spacing. 
But from a UI/UX perspective I feel like having a smaller gap between the device list and the header/button/timer kind of groups them together and is more intuitive.

File: packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx (hunk 3 above)
Comment (maximeflamant) [reply packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx:14]: From my debugging I found out that without the `memo`, each time `setTimer` is invoked it triggers a re-rendering of the whole device list, whereas with `memo` the device list will only be re-rendered when its props (`devices`) change.

Comment (t0bst4r) [reply packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx:14]: interesting. let's keep it like this, since it doesnt hurt.

**Review by t0bst4r — APPROVED** [review]

**Pull request merged**